build.rs

```diff
@@ -1,8 +1,14 @@
+use std::path::Path;
 use std::process::Command;
 use std::str::from_utf8;
 
 fn generate_self_schema() {
-    let output = Command::new("python")
+    if Path::new("./src/self_schema.py").exists() && option_env!("DEBIAN_FRONTEND") == Some("noninteractive") {
+        // self_schema.py already exists and DEBIAN_FRONTEND indicates we're in a maturin build,
```

Comment:
Checking `DEBIAN_FRONTEND` will cause `self_schema.py` not used when pip install from sdist I think?

Maybe it's better if we can leverage [`cargo:rerun-if-changed=PATH`](https://doc.rust-lang.org/cargo/reference/build-scripts.html#rerun-if-changed) to trigger rebuilds when `_types.py` changes.